### PR TITLE
update rules_boost to latest version

### PR DESCRIPTION
**Pull request by Wyverald**
Want to contribute? Great! Make sure you've read and understood
[CONTRIBUTING.md](https://github.com/cartographer-project/cartographer/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Wyverald** [comment]
gentle ping. Cartographer remains broken in the downstream pipeline, and if we can't resolve that soon, we'd need to disable Cartographer on Bazel CI.

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -20,11 +20,9 @@ def cartographer_repositories():
     _maybe(
         http_archive,
         name = "com_github_nelhage_rules_boost",
-        sha256 = "371f49e7b29e44a718baf8b9a2dd3eca865005a851c9ecf8fb6a10a715aa58dd",
-        strip_prefix = "rules_boost-a5a95642f6097f8949020646ffe89d7243008981",
+        strip_prefix = "rules_boost-c3fae06e819ed8b93e31b150387dce4864758643",
```
Comment (wohe) [inline bazel/repositories.bzl:23]: Can you add `sha256 = "f7d620c0061631d5b7685cd1065f2e2bf0768559555010a75e8e4720006f5867"`? Omitting it is a security risk, see https://bazel.build/rules/lib/repo/http.

**Review by wohe — COMMENTED** [review]
Other than the missing SHA256 this looks good to me. Once you add it, I'd be happy to merge.

File: bazel/repositories.bzl (hunk 1 above)
Comment (Wyverald) [reply bazel/repositories.bzl:24]: done.

**Review by wohe — APPROVED** [review]

**Pull request merged**